lib/hccrawler.js

```diff
@@ -284,14 +285,15 @@ class HCCrawler extends EventEmitter {
    */
   async _startRequest(options, depth) {
     const skip = await this._skipRequest(options);
-    await this._markRequested(options);
```

Comment:
it was not a good idea to mark requested immediately after starting request, because it will never be requested if the request is stopped in the middle. should be moved to https://github.com/yujiosaka/headless-chrome-crawler/pull/223/files#diff-1b8c3610c5c4374df25c5dd5f30362edR340